File: app/src/main/java/com/noxvision/app/data/MediaModels.kt
package com.noxvision.app.data

import android.net.Uri

data class CameraFile(
    val name: String,
    val size: Long,
    val date: String,
    val type: String
)

enum class GallerySource { CAMERA_DEVICE, PHONE }

enum class PhoneFolder { CAMERA, PICTURES, NOXVISION }

data class PhoneMediaFile(
    val uri: Uri,
    val name: String,
    val size: Long,
    val dateAddedSec: Long,
    val mime: String,
    val isVideo: Boolean
)

File: NoxVision-iOS/NoxVision/Models/MediaModels.swift
import Foundation

struct CameraFile: Identifiable {
    let id = UUID()
    let name: String
    let size: Int64
    let date: String
    let type: String
}

enum GallerySource: String, CaseIterable {
    case cameraDevice = "Thermal"
    case phone = "Phone"
}

enum PhoneFolder: String, CaseIterable {
    case camera = "Camera"
    case pictures = "Pictures"
    case noxvision = "NoxVision"
}

struct PhoneMediaFile: Identifiable {
    let id = UUID()
    let url: URL
    let name: String
    let size: Int64
    let dateAdded: Date
    let mimeType: String
    let isVideo: Bool
}
